### PR TITLE
Support to update credentials manually to an account

### PR DESCRIPTION
**Pull request by achouhan09**
### Explain the changes
1. Added a CLI command to manually update account credentials (access key and secret key). Users can provide the access key and secret key as flags, or they will be prompted to enter them if the flags are not specified.
For example: `noobaa account credentials --access-key=<access-key> --secret-key=<secret-key>`

Additionally, validation checks have been implemented for both keys to ensure the command succeeds only with valid credentials.

### Testing Instructions:
1. `make test-cli-flow`

- [X] Tests added

### CODE REVIEW

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:1]: There is the part of `cmd.AddCommand` and I don't see the `CmdCredentials` in it.
Try to run `noobaa account --help` and make sure you see the added command.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:413]: Maybe set the condition with a non-negative condition?
if we have CRD or if the name is admin@noobaa.io... (the code that you have in the "else" clause).

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:791]: typo - reply

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:93]: Sure, will do. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:1]: Forgot to add it, will add. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:791]: Sure, thanks.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:84]: You added in `CmdRegenerate` instead of `CmdCredentials`.
Anyway, I think you can remove it, because now I understand the whole menu is with "credential" as a known term, so no need to explain.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:763]: Why public function?

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:819]: Why public function?

Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:819]: I think there are mutual parts between the functions `UpdateNonCrdAccountKeys` and `UpdateAccountKeys`.
What about using one function and adding a variable for the case that you handle in `UpdateAccountKeys` and not in `UpdateNonCrdAccountKeys` (the secret)

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:412]: What did you think about adding a check here as well? We would have fewer RPC calls.
And also explain the credentials requirements.

**Comment by shirady** [comment]
@achouhan09, 
General question - At which stage will the user set the credentials manually to the account?

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:84]: Sure, will update it. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:819]: I am not sure about this, all the functions in this file are public because of there reusability and easy access.

Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:819]: Yes we can do that. I am following the same approach here(as we did for Regenerate). But its our call if we want to change it then we can change it. Also later we need to do the same for Regenerate I think.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:412]: Yes, I am thinking of the same. I will add here as well. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [reply pkg/noobaaaccount/noobaaaccount.go:819]: I understand, so you can keep it.

**Comment by achouhan09** [comment]
> @achouhan09, General question - At which stage will the user set the credentials manually to the account?

File: pkg/nb/types.go
Comment (shirady) [inline pkg/nb/types.go:434]: Do you see the access key and secret key printings in the operator logs after the RPC call?
Please notice PR #1353, in case you see this I think you should wait for the PR to be merged and use the same solution.
cc: @liranmauda

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:99]: I think you should warn the user that using the flag is not secure, for example:
https://github.com/noobaa/noobaa-operator/blob/fcc5240f5af690596823f2ab63e9debd9a640fda/pkg/namespacestore/namespacestore.go#L82

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:409]: What do you think about using `util.GetFlagStringOrPromptPassword` to avoid this printing in the shell? for example:
https://github.com/noobaa/noobaa-operator/blob/fcc5240f5af690596823f2ab63e9debd9a640fda/pkg/namespacestore/namespacestore.go#L447-L448

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:875]: To avoid the long list of variables in the function signature maybe use a struct.
Since you are passing the list of boolean, which is hard-coded, I think it would be better to use defaults like you have in core:
https://github.com/noobaa/noobaa-core/blob/ff0007ad2512b0f67efa80112ceb2fcea0f3a63a/src/util/string_utils.js#L226-L242

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:879]: Please notice that in the core repo, you are using `validate_complexity` which uses `validate_length` which checks `len(str) != length`, see:
https://github.com/noobaa/noobaa-core/blob/ff0007ad2512b0f67efa80112ceb2fcea0f3a63a/src/util/string_utils.js#L156-L161

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:883]: I think that creating a variable for the regex is more readable.
For example:
https://github.com/noobaa/noobaa-core/blob/ff0007ad2512b0f67efa80112ceb2fcea0f3a63a/src/util/string_utils.js#L11C2-L11C2

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:874]: I think you can add more description

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:875]: I think you can move this function to `util.go`, if you want to can also separate it for smaller functions like in the core repo:
https://github.com/noobaa/noobaa-core/blob/ff0007ad2512b0f67efa80112ceb2fcea0f3a63a/src/util/string_utils.js#L236-L240

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:99]: Make sense, will update it. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:409]: Yes, I think it is a good idea to get keys using prompt and also not printing them on terminal. Thanks, will update it.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:875]: yes I will move it there.

Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:875]: Sure, Thanks.

File: pkg/util/util.go
Comment (shirady) [inline pkg/util/util.go:106]: Please notice the changes in core in PR [noobaa/noobaa-core/#8008](https://github.com/noobaa/noobaa-core/pull/8008), see here:
https://github.com/noobaa/noobaa-core/blob/56e17689c9937ee91f5516b8ed73f1ceb1eee6e5/src/util/string_utils.js#L11-L12

Comment (achouhan09) [reply pkg/util/util.go:106]: Thanks, I will update core PR accordingly.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go]: Updated the code to use a single regex expression instead of multiple functions, just like noobaa-core.Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go]: Updated. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:94]: Done.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:99]: Sure. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:103]: Sure, Thanks.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:870]: Why did you prefer to use `MustCompile` and not `Compile`?
As I understand it `Compile` (in this case `MustCompile`) should be run in the "init" stage and then you can reuse the regexp, in this function you compile and recreate it on every call ([here is a simple example](https://pkg.go.dev/regexp#example-package)).

File: pkg/noobaaaccount/noobaaaccount.go
Comment (shirady) [inline pkg/noobaaaccount/noobaaaccount.go:865]: In the core repo the term "complexity" was removed.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:870]: Yes, It should be the better way. Thanks will update it.

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:865]: I will remove it from operator. Thanks

File: pkg/nb/types.go
Comment (achouhan09) [reply pkg/nb/types.go:434]: I cannot see any of them(access_key and secret_key) printing in operator logs and in core logs printed as some sensitive string. I think we need not to make changes in code for this. Thanks

Comment (shirady) [reply pkg/nb/types.go:434]: As discussed they currently appear as printing of the RPC request details in the CLI.

Comment (liranmauda) [reply pkg/nb/types.go:434]: Great Catch,
It is unrelated to this PR.
@shirady Could you open an Issue for it? 
@achouhan09 We can proceed with this PR

Comment (shirady) [reply pkg/nb/types.go:434]: @liranmauda sure
@achouhan09 resolving according to the comment above.

Comment (liranmauda) [reply pkg/nb/types.go:434]: https://github.com/noobaa/noobaa-operator/pull/1357 should address this issue

Comment (achouhan09) [reply pkg/nb/types.go:434]: Thanks Liran, Should I need to update the type to `MaskedString` for accessKeys?

Comment (shirady) [reply pkg/nb/types.go:434]: After PR #1357 will be merged it will work, you will not need to change something in your code.

**Review by shirady — APPROVED** [review]
LGTM

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:406]: Make sense, I will update it. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:423]: I think it will take lots of code changes as we have used it in various places, will do it in separate PR if required. Thanks

File: pkg/noobaaaccount/noobaaaccount.go
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go:816]: Not relevant to this PR, will do it in another PR. Thanks.

**Pull request merged**
